MobiusCore/Source/EffectHandlers/EffectHandler.swift

```diff
@@ -17,16 +17,47 @@
 // specific language governing permissions and limitations
 // under the License.
 
-// Primitive EffectHandler
-public struct EffectHandler<Effect, Event> {
-    public let handle: (Effect, @escaping Consumer<Event>) -> Void
-    public let disposable: Disposable
+/// This protocol defines the contract for an Effect Handler which takes `Effect`s as input, and produces `Event`s as output.
+/// For each incoming `Effect`, zero or more `Event`s can be sent as output using `callback.send(Event)`.
+/// When an `Effect` has been completely handled, i.e. that effect will not result in any more events, call `callback.end()`.
+///
+/// Note: `EffectHandler` should be used in conjunction with an `EffectRouter`.
+public protocol EffectHandler {
+    associatedtype Effect
+    associatedtype Event
+
+    /// Handle an `Effect`.
+    /// To output events, call `callback.send`.
+    /// When you are done handling `input`, be sure to call `callback.end()` to prevent memory leaks.
+    /// If it does not make sense to finish handling an effect, you should be using a `Connectable` instead of this protocol.
```

Comment:
What kind of cases do you see this applying to?

Reply:
@pettermahlen and I had a discussion about this, so maybe he can provide some insight too.

The way I see it, `Connectable` is still useful in the cases where a clear connection between effects and events cannot be made. For example, an effect handler which is used as an adapter to talk between two features. Such an effect handler would translate effects from one loop into messages to the other feature, and relay messages from the other feature back into the loop.